clients/src/main/java/org/apache/kafka/clients/admin/DeleteTopicsResult.java

```diff
@@ -30,24 +31,62 @@
  */
 @InterfaceStability.Evolving
 public class DeleteTopicsResult {
-    final Map<String, KafkaFuture<Void>> futures;
+    private Map<String, KafkaFuture<Void>> nameFutures;
+    private Map<Uuid, KafkaFuture<Void>> topicIdFutures;
 
-    protected DeleteTopicsResult(Map<String, KafkaFuture<Void>> futures) {
-        this.futures = futures;
+    protected DeleteTopicsResult() {}
```

Comment:
Could this be private? I think it would be a little cleaner to take `nameFutures` and `topicIdFutures` in the constructor. That would allow us to mark those fields as final.

Reply:
So the idea is that we have two parameters and set one as null in the static methods?

Reply:
Yeah, right. As long as the constructor is private, you can ensure one of them must be null.

Reply:
Ah I remember the issue with that being private. There is a test that extends this class.  

> @hachikuji I originally wanted the constructor for DeleteTopicsResult to be private, but InternalTopicManagerTest required creating a subclass.

https://github.com/apache/kafka/pull/10892#issuecomment-862876282

The issue with just using the static method is that it is protected in the class. So the options are to either make the class protected (as is now), make the static method public, or change the test.